test/unit/math/prim/arr/err/is_nonzero_size_test.cpp

```diff
@@ -0,0 +1,35 @@
+#include <stan/math/prim/arr.hpp>
+#include <gtest/gtest.h>
+#include <test/unit/util.hpp>
+#include <limits>
+#include <vector>
+
+TEST(ErrorHandlingMatrix, IsNonzeroSizeMatrix) {
+  using stan::math::is_nonzero_size;
+
+  std::vector<double> a;
```

Comment:
This should be

```
std::vector<double> a = {3, 3, 3, 3};
```

or you can use the dedicated duplication constructor,

```
std::vector<double> a(4, 3);
```

In general, we don't need `3.0` when the argument type is `double`---the compiler's smart enough to figure this all out statically.